File: sdkwork-app-sdk-swift/Sources/API/CartApi.swift
import Foundation

public class CartApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// 更新购物车商品数量
    public func updateItemQuantity(itemId: String, body: UpdateCartItemForm) async throws -> PlusApiResultCartItemVO? {
        let response = try await client.put(ApiPaths.appPath("/cart/items/\(itemId)"), body: body)
        return response as? PlusApiResultCartItemVO
    }

    /// 删除购物车商品
    public func removeItem(itemId: String) async throws -> PlusApiResultVoid? {
        let response = try await client.delete(ApiPaths.appPath("/cart/items/\(itemId)"))
        return response as? PlusApiResultVoid
    }

    /// 更新商品选中状态
    public func updateItemSelection(itemId: String, params: [String: Any]? = nil) async throws -> PlusApiResultVoid? {
        let response = try await client.put(ApiPaths.appPath("/cart/items/\(itemId)/select"), body: nil, params: params)
        return response as? PlusApiResultVoid
    }

    /// 批量更新选中状态
    public func batchUpdateSelection(body: BatchSelectForm) async throws -> PlusApiResultVoid? {
        let response = try await client.put(ApiPaths.appPath("/cart/items/select"), body: body)
        return response as? PlusApiResultVoid
    }

    /// 获取购物车商品列表
    public func getCartItems() async throws -> PlusApiResultListCartItemVO? {
        let response = try await client.get(ApiPaths.appPath("/cart/items"))
        return response as? PlusApiResultListCartItemVO
    }

    /// 添加商品到购物车
    public func addItem(body: AddCartItemForm) async throws -> PlusApiResultCartItemVO? {
        let response = try await client.post(ApiPaths.appPath("/cart/items"), body: body)
        return response as? PlusApiResultCartItemVO
    }

    /// 批量删除购物车商品
    public func removeItems(params: [String: Any]? = nil) async throws -> PlusApiResultVoid? {
        let response = try await client.delete(ApiPaths.appPath("/cart/items"), params: params)
        return response as? PlusApiResultVoid
    }

    /// 获取当前用户购物车
    public func getMy() async throws -> PlusApiResultShoppingCartVO? {
        let response = try await client.get(ApiPaths.appPath("/cart"))
        return response as? PlusApiResultShoppingCartVO
    }

    /// 清空购物车
    public func clear() async throws -> PlusApiResultVoid? {
        let response = try await client.delete(ApiPaths.appPath("/cart"))
        return response as? PlusApiResultVoid
    }

    /// 获取购物车统计
    public func getCartStatistics() async throws -> PlusApiResultCartStatisticsVO? {
        let response = try await client.get(ApiPaths.appPath("/cart/statistics"))
        return response as? PlusApiResultCartStatisticsVO
    }

    /// 获取选中的商品列表
    public func getSelectedItems() async throws -> PlusApiResultListCartItemVO? {
        let response = try await client.get(ApiPaths.appPath("/cart/items/selected"))
        return response as? PlusApiResultListCartItemVO
    }

    /// 检查商品是否在购物车中
    public func checkItemIn(params: [String: Any]? = nil) async throws -> PlusApiResultBoolean? {
        let response = try await client.get(ApiPaths.appPath("/cart/check"), params: params)
        return response as? PlusApiResultBoolean
    }
}

File: sdkwork-app-sdk-csharp/Api/CartApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using App.Http;
using App.Models;

namespace App.Api
{
    public class CartApi
    {
        private readonly HttpClient _client;

        public CartApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// 更新购物车商品数量
        /// </summary>
        public async Task<PlusApiResultCartItemVO?> UpdateItemQuantityAsync(string itemId, UpdateCartItemForm body)
        {
            return await _client.PutAsync<PlusApiResultCartItemVO>(ApiPaths.AppPath($"/cart/items/{itemId}"), body);
        }

        /// <summary>
        /// 删除购物车商品
        /// </summary>
        public async Task<PlusApiResultVoid?> RemoveItemAsync(string itemId)
        {
            return await _client.DeleteAsync<PlusApiResultVoid>(ApiPaths.AppPath($"/cart/items/{itemId}"));
        }

        /// <summary>
        /// 更新商品选中状态
        /// </summary>
        public async Task<PlusApiResultVoid?> UpdateItemSelectionAsync(string itemId, Dictionary<string, object>? query = null)
        {
            return await _client.PutAsync<PlusApiResultVoid>(ApiPaths.AppPath($"/cart/items/{itemId}/select"), null, query);
        }

        /// <summary>
        /// 批量更新选中状态
        /// </summary>
        public async Task<PlusApiResultVoid?> BatchUpdateSelectionAsync(BatchSelectForm body)
        {
            return await _client.PutAsync<PlusApiResultVoid>(ApiPaths.AppPath("/cart/items/select"), body);
        }

        /// <summary>
        /// 获取购物车商品列表
        /// </summary>
        public async Task<PlusApiResultListCartItemVO?> GetCartItemsAsync()
        {
            return await _client.GetAsync<PlusApiResultListCartItemVO>(ApiPaths.AppPath("/cart/items"));
        }

        /// <summary>
        /// 添加商品到购物车
        /// </summary>
        public async Task<PlusApiResultCartItemVO?> AddItemAsync(AddCartItemForm body)
        {
            return await _client.PostAsync<PlusApiResultCartItemVO>(ApiPaths.AppPath("/cart/items"), body);
        }

        /// <summary>
        /// 批量删除购物车商品
        /// </summary>
        public async Task<PlusApiResultVoid?> RemoveItemsAsync(Dictionary<string, object>? query = null)
        {
            return await _client.DeleteAsync<PlusApiResultVoid>(ApiPaths.AppPath("/cart/items"), query);
        }

        /// <summary>
        /// 获取当前用户购物车
        /// </summary>
        public async Task<PlusApiResultShoppingCartVO?> GetMyAsync()
        {
            return await _client.GetAsync<PlusApiResultShoppingCartVO>(ApiPaths.AppPath("/cart"));
        }

        /// <summary>
        /// 清空购物车
        /// </summary>
        public async Task<PlusApiResultVoid?> ClearAsync()
        {
            return await _client.DeleteAsync<PlusApiResultVoid>(ApiPaths.AppPath("/cart"));
        }

        /// <summary>
        /// 获取购物车统计
        /// </summary>
        public async Task<PlusApiResultCartStatisticsVO?> GetCartStatisticsAsync()
        {
            return await _client.GetAsync<PlusApiResultCartStatisticsVO>(ApiPaths.AppPath("/cart/statistics"));
        }

        /// <summary>
        /// 获取选中的商品列表
        /// </summary>
        public async Task<PlusApiResultListCartItemVO?> GetSelectedItemsAsync()
        {
            return await _client.GetAsync<PlusApiResultListCartItemVO>(ApiPaths.AppPath("/cart/items/selected"));
        }

        /// <summary>
        /// 检查商品是否在购物车中
        /// </summary>
        public async Task<PlusApiResultBoolean?> CheckItemInAsync(Dictionary<string, object>? query = null)
        {
            return await _client.GetAsync<PlusApiResultBoolean>(ApiPaths.AppPath("/cart/check"), query);
        }
    }
}
